src/sublime_aio.py

```diff
@@ -1,20 +1,36 @@
 from __future__ import annotations
 import asyncio
 import atexit
+import functools
 import traceback
 from inspect import iscoroutinefunction
 from threading import Thread
-from typing import TYPE_CHECKING
-
-if TYPE_CHECKING:
-    from asyncio import Task
-    from concurrent.futures import Future
-    from typing import Any, Callable, Coroutine
+from typing import TYPE_CHECKING, overload
 
 import sublime
 import sublime_api
 import sublime_plugin
 
+if TYPE_CHECKING:
+    from collections.abc import Coroutine
+    from concurrent.futures import Future
+    from typing import Any, Callable, ParamSpec, TypeVar, TypeAlias
+
+    P = ParamSpec('P')
+    T = TypeVar('T')
+
+    BlankCoro: TypeAlias = Coroutine[object, None, None]
+    CompletionsReturnVal: TypeAlias = (
+        sublime.CompletionList
+        | tuple[list[sublime.CompletionValue], sublime.AutoCompleteFlags]
+        | list[sublime.CompletionValue]
+        | None
+    )
```

Comment:
This is what `sublime_types.CompletionsListValue` should be used for, but it doesn't seem to be as correct as this one. Maybe we suggest a fix.

Reply:
I would agree, but `sublime_types` is not part of the LSP stubs and the `CompletionValue` that they define is not exported, so I resorted to just re-defining it.

Reply:
Yeah, actually this is something I've defined downstream in my patched versions of ST APIs. The comment rather targeted into direction of proposing to add it to sublime_types and export it as various listeners make use of it.